tap_bitbucket/gitlocal.py

```diff
@@ -199,11 +232,25 @@ def _cloneRepo(self, repo, repoWdir):
         raise GitLocalException("Clone of repo {} failed with code {}, message: {}"\
           .format(repo, completed.returncode, strippedOutput))
 
+  def _initRootCommits(self, repo, repoWdir):
```

Comment:
I'm not a huge fan of having to do this when we init the repo. We need this information when we read in the diff data. We use multiple threads to do that work and I didn't want to bother making this thread safe. 

Previously, it was easy to read the returncode of the process and check the output for a particular string, which we would use to denote a root commit. Now that we are streaming stdout from the `git diff` command, it gets more difficult to do that without sleeping to wait a few seconds for the command to fail.